Code/Mantid/Framework/WorkflowAlgorithms/src/ConvolutionFitSequential.cpp

```diff
@@ -0,0 +1,670 @@
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/StringContainsValidator.h"
+
+#include <cmath>
+
+namespace {
+Mantid::Kernel::Logger g_log("ConvolutionFitSequential");
+}
+
+namespace Mantid {
+namespace Algorithms {
+
+using namespace API;
+using namespace Kernel;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(ConvolutionFitSequential)
+
+//----------------------------------------------------------------------------------------------
+/** Constructor
+ */
+ConvolutionFitSequential::ConvolutionFitSequential() {}
+
+//----------------------------------------------------------------------------------------------
+/** Destructor
+ */
+ConvolutionFitSequential::~ConvolutionFitSequential() {}
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string ConvolutionFitSequential::name() const {
+  return "ConvolutionFitSequential";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int ConvolutionFitSequential::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string ConvolutionFitSequential::category() const {
+  return "Workflow\\MIDAS";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string ConvolutionFitSequential::summary() const {
+  return "Performs a sequential fit for a convolution workspace";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void ConvolutionFitSequential::init() {
+  declareProperty(
+      new WorkspaceProperty<>("InputWorkspace", "", Direction::Input),
+      "The input workspace for the fit.");
+
+  auto scv = boost::make_shared<StringContainsValidator>();
+  auto requires = std::vector<std::string>();
+  requires.push_back("Convolution");
+  requires.push_back("Resolution");
+  scv->setRequiredStrings(requires);
+
+  declareProperty("Function", "", scv,
+                  "The function that describes the parameters of the fit.",
+                  Direction::Input);
+
+  std::vector<std::string> backType;
+  backType.push_back("Fixed Flat");
+  backType.push_back("Fit Flat");
+  backType.push_back("Fit Linear");
+
+  declareProperty("BackgroundType", "Fixed Flat",
+                  boost::make_shared<StringListValidator>(backType),
+                  "The Type of background used in the fitting",
+                  Direction::Input);
+
+  declareProperty(
+      "StartX", EMPTY_DBL(), boost::make_shared<MandatoryValidator<double>>(),
+      "The start of the range for the fit function.", Direction::Input);
+
+  declareProperty(
+      "EndX", EMPTY_DBL(), boost::make_shared<MandatoryValidator<double>>(),
+      "The end of the range for the fit function.", Direction::Input);
+
+  declareProperty("Temperature", EMPTY_DBL(),
```

Comment:
It seems all this is doing is adding the sample log, this should just be done in the UI code as currently it gives the impression the algorithm is responsible for the correction.